src/ray/raylet/main.cc

```diff
@@ -63,6 +63,7 @@ DEFINE_string(resource_dir, "", "The path of this ray resource directory.");
 DEFINE_int32(ray_debugger_external, 0, "Make Ray debugger externally accessible.");
 // store options
 DEFINE_int64(object_store_memory, -1, "The initial memory of the object store.");
+DEFINE_string(node_name, "", "The user-provided identifier or name for this node.");
```

Comment:
Should we set a default node name as the IP address or NodeID or something? I imagine if we really want to use this everywhere, that would greatly simplify the logic everywhere.